pkg/protocol/engine/engine.go

```diff
@@ -188,15 +193,18 @@ func New(
 			} else {
 				// Restore from Disk
 				e.Storage.Prunable.RestoreFromDisk()
-
 				e.EvictionState.PopulateFromStorage(e.Storage.Settings().LatestCommitment().Index())
+
 				if err := e.Attestations.RestoreFromDisk(); err != nil {
 					panic(ierrors.Wrap(err, "failed to restore attestations from disk"))
 				}
 				if err := e.UpgradeOrchestrator.RestoreFromDisk(e.Storage.Settings().LatestCommitment().Index()); err != nil {
 					panic(ierrors.Wrap(err, "failed to restore upgrade orchestrator from disk"))
 				}
+
+				e.startupMaxBlockSlot = e.Storage.Settings().LatestCommitment().Index() + e.CurrentAPI().ProtocolParameters().MaxCommittableAge()
```

Comment:
Maybe add a comment here that this represent your "caching" window for blocks, below which blocks cannot be found in cache and should be found in storage instead. I would also rename this field to something more in line with what it actually does.

Reply:
What name would you suggest?

Reply:
Added a comment and renamed to `startupAvailableBlocksWindow`

Reply:
Maybe `cachedBlocksWindow`?

Reply:
I don't think they are cached. They are accepted blocks from the previous "run" and only available if you restart your node. So I think what I have now `startupAvailableBlocksWindow` fits better